tests/arc/tdup.nim

```diff
@@ -0,0 +1,53 @@
+discard """
+  cmd: "nim c --mm:arc --expandArc:foo --hints:off $file"
+  nimout: '''
+--expandArc: foo
+
+var
+  x
+  :tmpD
+  s
+  :tmpD_1
+x = Ref(id: 8)
+inc:
+  `=dup`(:tmpD, x)
+  :tmpD
+inc:
+  let blitTmp = x
+  blitTmp
+s = RefCustom(id_1: 777)
+inc_1 :
+  :tmpD_1 = `=dup`(s)
+  :tmpD_1
+inc_1 :
+  let blitTmp_1 = s
+  blitTmp_1
+-- end of expandArc ------------------------
+'''
+"""
+
+type
+  Ref = ref object
+    id: int
+
+  RefCustom = ref object
+    id: int
+
+proc inc(x: sink Ref) =
+  doAssert x.id == 8
+
+proc inc(x: sink RefCustom) =
+  doAssert x.id == 777
+
+proc `=dup`(x: RefCustom): RefCustom =
```

Comment:
What is going on here? RefCustom doesn't support a custom =dup operation.

Reply:
I'm most likely missing something. Does `=dup` support being overridden? 

Or Is there a way to distinguish `Ref` from `ref Object`?
```nim
proc `=dup`(x: Ref): Ref 
proc `=dup`(x: ref Object): ref Object
```

Reply:
> Does =dup support being overridden?

Yes, but not for `ref` types. It works just like the other hooks which cannot be overriden for `ref` either.

Reply:
I see, thanks!